modules/sources/contracts/v3/fetch-pool-data.ts

```diff
@@ -87,26 +97,54 @@ export async function fetchPoolData(
                 ? (results[pointer + 3].result as unknown as PoolTokenRates)
                 : undefined;
 
+        const totalShares = formatEther(
+            results[pointer].status === 'success' ? (results[pointer].result as bigint) : 0n,
+        );
+
+        const decimals = decodeDecimalDiffs(Number(config?.tokenDecimalDiffs), poolTokenInfo?.[0].length ?? 0);
```

Comment:
is that basically the scaling factor?

Reply:
these are token decimals, 18, 6 etc - they just store it compressed in one int